stored_requests/caches/in_memory/lru.go

```diff
@@ -13,8 +13,8 @@ import (
 // These prefixes are used when generating cache keys so that there's no chance
 // of conflict between the two types.
 const (
```

Comment:
yuck... you're totally right, again. This would work with `uuid` (which is fixed length) or `int` (which doesn't include letters)`... but does fail on general strings.

I was trying to avoid two caches... not really for the sake of performance, but more because you need to define the cache size up front. If we've got 1 GB to work with, there's no good way for us to guess how much of this is "stored request" space vs. "stored imp" space.

I could go either way as well... but correctness definitely is generally much more important than performance.

For now I think I'll naively divide the configured cache space in half, since this PR is already pretty big. In the future we should probably add another config option to allow more control.